collective-big-count/Makefile

```diff
@@ -0,0 +1,109 @@
+#
+#  Copyright (c) 2021-2022 IBM Corporation.  All rights reserved.
+#
+# $COPYRIGHT$
+#
+
+######################################################################
+# Utilities
+######################################################################
+.PHONY: default help
+
+CC	= mpicc
+F77	= mpif77
+F90	= mpif90
+MPIRUN = mpirun
+RM	= /bin/rm -f
+
+# GCC
+CC_FLAGS = -g -O0 -Wall -Werror
+# Clang
+#CC_FLAGS = -g -O0 -Wall -Wshorten-64-to-32 -Werror
+F90_FLAGS =
+F77_FLAGS = $(F90_FLAGS)
+
+######################################################################
+# TEST_UNIFORM_COUNT: (Default defined here)
+#  The 'count' size to be used regardless of the datatype
+#  This should never exceed that of INT_MAX (2147483647) which
+#  is the maximum count allowed by the MPI Interface in MPI 3
+######################################################################
+# Test at the limit of INT_MAX : 2147483647
+TEST_UNIFORM_COUNT=2147483647
+
+######################################################################
+# TEST_PAYLOAD_SIZE: (Default in common.h)
+#  This value is the total payload size the collective should perform.
+#  The 'count' is calculated as relative to the datatype size so
+#  as to target this payload size as closely as possible:
+#  count = TEST_PAYLOAD_SIZE / sizeof(datatype)
+######################################################################
+# INT_MAX    : == 2 GB so guard will not trip
+TEST_PAYLOAD_SIZE=2147483647
```

Comment:
You can, but it should be relative to the datatype. So we don't want to set it to `INT_MAX x 8` because then the count would be too big for an `int` datatype. I left it as a default of `INT_MAX` and folks can change it as they see fit. The `TEST_UNIFORM_COUNT` is probably the more useful of the two variants.